src/syntax/ast.rs

```diff
@@ -1620,7 +1620,11 @@ pub enum DestructuringKind {
 impl Pattern {
     /// The kind of the pattern.
     pub fn kind(&self) -> PatternKind {
-        if self.0.children().len() <= 1 {
+        let has_sink =
+            self.0.children().find(|c| c.kind() == SyntaxKind::Spread).is_some();
+
+        // Check if pattern only has one child and two parenthesese and is not a sink
+        if self.0.children().len() <= 3 && !has_sink {
```

Comment:
is this robust? not sure how this works exactly, but could the number of children be whitespace dependent?

Reply:
For `()` it fails, but that should not matter, we get the default identifier either as `PatternKind::Ident` or `PatternKind::Destructure([PatternKind::Ident])`. For `(a: b)` the children are two parentheses and one `Named` so it seems to work.

Reply:
This test fails for `let (x       ) = 3`